search/includes/classes/queue/class-cron.php

```diff
@@ -114,6 +120,32 @@ public function process_jobs( $job_ids ) {
 		$this->queue->process_jobs( $jobs );
 	}
 
+	/**
+	 * Given a term id, queue all posts for reindexing that match it
+	 *
+	 * @param {int} $term_taxonomy_id The term id you want to fetch/index
+	 */
+	public function queue_posts_for_term_taxonomy_id( $term_taxonomy_id ) {
+		error_log( 'Taxonomy='.$term_taxonomy_id );
+		if ( ! is_int( $term_taxonomy_id ) ) {
+			return;
+		}
+
+		global $wpdb;
+
+		$query = $wpdb->prepare( 'SELECT `object_id` FROM `' . $wpdb->term_relationships . '` WHERE `term_taxonomy_id` = %d', $term_taxonomy_id );
```

Comment:
Two things - we should batch this so we process more manageable chunks at a time (loading 275k+ rows at once probably won't work). We can also just use a WP_Query() to do this (vs direct DB query).

Reply:
Made the switch, still have to figure out batching with WP_Query.